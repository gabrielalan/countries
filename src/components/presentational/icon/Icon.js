import React from 'react';

export default function Icon({ id, className }) {
  return <span className={'icon ' + (className || '')}>
    <svg viewBox="0 0 10 10" className="icon__svg" aria-hidden="true">
      <use xlinkHref={`#icon-${id}`} />
    </svg>
  </span>
}