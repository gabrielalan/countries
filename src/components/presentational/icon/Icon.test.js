import React from 'react';
import renderer from 'react-test-renderer';
import Icon from './Icon';

describe('Icon', () => {
  it('should render svg normally', () => {
    const tree = renderer.create(<Icon id="clear" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should allow classNames', () => {
    const tree = renderer.create(<Icon id="clear" className="some-class" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
