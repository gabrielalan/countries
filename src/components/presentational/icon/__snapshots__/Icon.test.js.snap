// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should allow classNames 1`] = `
<span
  className="icon some-class"
>
  <svg
    aria-hidden="true"
    className="icon__svg"
    viewBox="0 0 10 10"
  >
    <use
      xlinkHref="#icon-clear"
    />
  </svg>
</span>
`;

exports[`Icon should render svg normally 1`] = `
<span
  className="icon "
>
  <svg
    aria-hidden="true"
    className="icon__svg"
    viewBox="0 0 10 10"
  >
    <use
      xlinkHref="#icon-clear"
    />
  </svg>
</span>
`;
